src/sync/Synchroniser.js

```diff
@@ -56,52 +57,77 @@ export class Synchroniser {
     // initialisation, in which case we are continuing a failed partial initialisation. If the
     // serverURL is different, it is either completely fresh, or the URL has been changed so we
     // should start afresh
-    const isFresh = serverURL !== this.serverURL;
-    if (isFresh) this.database.write(() => { this.database.deleteAll(); });
+    const oldUrl = this.settings.get(SYNC_URL);
+    const isFresh = !oldUrl || serverURL !== oldUrl;
```

Comment:
#455 issue created for this, so we can get this PR going along while I resolve issue that before release.